src/Operators/Observables.jl

```diff
@@ -27,22 +35,43 @@ function Base.show(io::IO, ::MIME"text/plain", obs::ObservableInfo)
     printstyled(io,"")
 end
 
+function magnetization_observables(g,N)
+    dipoles(single_g,N) = if N == 0
+        dipole_component(i) = FunctionMap{Float64}(s -> (single_g*s)[i],1,3)
```

Comment:
Yes, `dx = dipole_component(1)` will be a 1x3 linear map, which you can just use as a matrix. When it acts, `dx * S` on a vector `S`, it will do `(s -> (single_g*s)[1])(S)` which becomes `(single_g * S)[1]`. So it's a lot like just taking `single_g[1,:]`, and maybe we could try that instead and compare performance. The reason to do it as a LinearMap is that you might be able to implement this faster than a matrix multiply! For example if `single_g` is `I(3)`, it may be optimized to be faster